src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -279,7 +281,7 @@ public T WithLegacyCacheCompatibility(bool enableLegacyCacheCompatibility = true
             return (T)this;
         }
 
-#if !XAMARINMAC2_0
+#if !XAMARINMAC2_0 !ANDROID && !iOS
```

Comment:
This is going to be hell to maintain :(. Not a fan.

Reply:
What is the error you are getting from the automated tests? Can we find a way to fix that instead?